File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    length = len(prices)
    for idx in range(length):
        tmp = 0
        for i in range(idx + 1, length):
            tmp += 1
            if prices[idx] > prices[i]:
                break
        answer.append(tmp)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int n = prices.length;
        int[] answer = new int[n];
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            while (!stack.isEmpty() && prices[stack.peek()] > prices[i]) {
                int top = stack.pop();
                answer[top] = i - top;
            }
            stack.push(i);
        }

        while (!stack.isEmpty()) {
            int top = stack.pop();
            answer[top] = n - 1 - top;
        }

        return answer;
    }
}